asset/js/widget/BaseInput.js

```diff
@@ -812,6 +831,20 @@ define(["../notjQuery", "Completer"], function ($, Completer) {
             }
         }
 
+        onTermClick(event) {
+            if (this.isReadOnlyMode()) {
+               this.removeTerm(event.target.parentNode);
+            }
+        }
+
+        isReadOnlyMode() {
+            return this.input.dataset.termMode === 'read-only'
+        }
+
+        isTermDirectionVertical() {
+            return this.input.dataset.termDirection === 'vertical'
+        }
```

Comment:
Don't place these methods here, there are only event listeners and this section is specifically for such.